activesupport/lib/active_support/core_ext/hash/compact.rb

```diff
@@ -1,23 +1,27 @@
 class Hash
-  # Returns a hash with non +nil+ values.
-  #
-  #   hash = { a: true, b: false, c: nil }
-  #   hash.compact        # => { a: true, b: false }
-  #   hash                # => { a: true, b: false, c: nil }
-  #   { c: nil }.compact  # => {}
-  #   { c: true }.compact # => { c: true }
-  def compact
-    select { |_, value| !value.nil? }
+  unless self.instance_methods(false).include?(:compact)
```

Comment:
```
class Hash
  def puts_self
    puts "Self#{self.class}"
  end
end


class HashImp < Hash

end


HashImp.new.puts_self

=> SelfHashImp
```
The self picks up the current class. Again, not sure about the implications, its better to be explicit.